lib/pure/os.nim

```diff
@@ -102,9 +98,59 @@ proc normalizePathEnd(path: var string, trailingSep = false) =
     # // => / (empty case was already taken care of)
     path = $DirSep
 
-proc normalizePathEnd(path: string, trailingSep = false): string =
-  result = path
-  result.normalizePathEnd(trailingSep)
+when (NimMajor, NimMinor) >= (1, 1):
+  proc normalizePathEnd*(path: var string, trailingSep = false) =
+    ## Ensures ``path`` has exactly 0 or 1 trailing `DirSep`, depending on
+    ## ``trailingSep``, and taking care of edge cases: it preserves whether
+    ## a path is absolute or relative, and makes sure trailing sep is `DirSep`,
+    ## not `AltSep`.
+    ## See also:
+    ## * `normalizePathEnd proc <#normalizePathEnd,string,bool>`_ for a version
+    ## which returns a new string.
+    normalizePathEndImpl(path, trailingSep = trailingSep)
+
+  proc normalizePathEnd*(path: string, trailingSep = false): string =
+    ## A version of ``normalizePathEnd()``
+    ## See also:
+    ## * `normalizePathEnd proc <#normalizePathEnd,string,bool>`_ for the
+    ## in-place version.
+    runnableExamples:
+      when defined(posix):
+        assert normalizePathEnd("/lib/", trailingSep = true) == "/lib/"
+        assert normalizePathEnd("", trailingSep = true) == "/"
+        assert normalizePathEnd("/lib", trailingSep = false) == "/lib"
+        assert normalizePathEnd("lib//", trailingSep = false) == "lib"
+        assert normalizePathEnd("lib", trailingSep = false) == "lib"
+        assert normalizePathEnd("/lib//", trailingSep = true) == "/lib/"
+    result = path
+    result.normalizePathEnd(trailingSep)
+
+else:
+  proc normalizePathEnd(path: var string, trailingSep = false) =
+    ## Ensures ``path`` has exactly 0 or 1 trailing `DirSep`, depending on
```

Comment:
i really don't like code duplication and we don't need runnableExamples + unittest for this un-exported version, all this is already in `normalizePathEnd*`